### PR TITLE
Try build/test different geese in CI

### PR DESCRIPTION
**Pull request by evalexpr**
First pass at #37 without support for cross-compilation of tests

Signed-off-by: evalexpr <wilkinsphysics@gmail.com>

### CODE REVIEW

**Comment by bradfitz** [comment]
Thanks. I guess this should wait until there's a GOOS besides Linux that passes.

**Comment by evalexpr** [comment]
@bradfitz Darwin seems to pass at least, I'll keep this open in the meantime.

I'm planning on writing the shell wrapper for the cross-compilation of tests, I'll push it up in a sec, but it again fails on Windows due to the same error.

Would appreciate some thoughts on it - also not sure if we want to split out the actions a bit, or are we happy to have them all as part of linux.yml?

**Comment by danderson** [comment]
Well, good news, the test failure is legit! 7e5e3277 changed an API on some platforms only.

This is exactly what I wanted to catch by adding this to CI, there's a thinko in the darwin build that makes it straight up not compile.

I've fixed the darwin/windows/openbsd builds at HEAD (in theory), can you rebase and let CI run again?

Also, your second commit is missing a Signed-Off-By, which is making DCO bot sad.

**Comment by evalexpr** [comment]
Rebased. Looks good now on Darwin but Windows is still failing due to #42  - I think after that is fixed we'll be good to go.

Meanwhile, I've added openbsd to the test script too - passes locally for me, but doesn't get a chance to run in the action due to the earlier failure when building for Windows.

I'll add back the Slack integration in the action after #42 is fixed.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -28,20 +28,21 @@ jobs:
...
 
-    - name: Test build
-      run: go test ./...
+    - name: macOS build
+      env:
+        GOOS: darwin
+        GOARCH: amd64
+      run: go build ./cmd/...
 
-    - uses: k0kubun/action-slack@v2.0.0
```
Comment (bradfitz) [inline .github/workflows/linux.yml]: Why is this being deleted?

Comment (danderson) [reply .github/workflows/linux.yml]: Temporary, to not spam our slack.

#42 should be fixed now, so a rebase + rerun should do the trick.

Comment (evalexpr) [reply .github/workflows/linux.yml]: Looks green all around now. I've added this back in.

**Comment by danderson** [comment]
\o/ all green. Thank you!

**Pull request merged**